src/prototype.rs

```diff
@@ -144,7 +144,38 @@ impl<'a> FirstPass<'a> {
             return self.parse_indented_code_block(ix, remaining_space);
         }
 
+
+        // HTML Blocks
+
+        // HTML blocks may have 1-3 spaces in front of them, but our scanning
+        // functions don't know that. The spaces are preserved in the output,
+        // so we don't want to move `ix` yet.
         line_start.scan_all_space();
+        let nonspace_ix = start_ix + line_start.bytes_scanned();
+
+        // HTML block openings always happen on a single line, but our scanning
```

Comment:
The scanning functions should never consume past the end of a line, I think. The first pass parses the input into lines, and the resulting items in the chain are only within a line (with the partial exception of SoftBreak and HardBreak). Inline HTML scanning happens from an InlineScanner which only sees inside the items produced in the first pass. It's entirely possible there are cases this isn't strictly true, but those are probably bugs and maybe the scanners should be changed. In any case, this comment is likely to be confusing.

It is true the scanners only need to see to the end of the line, so I'm not particularly worried the code is wrong.

Reply:
I don't think `scan_attribute_value` is being used for inline HTML. It's all being done by `scan_inline_html` which in this case calls into `scan_inline_attribute`. Both of those use an `InlineScanner` struct, not the source text.

I think the goal here is that old-style scanners (ie not those using InlineScanner) can be used to scan block structure, and they should always stop at the end of a line. They should be separate from the inline case (there might be a bit of logic duplication, but I think there will be important subtle differences anyway). Does that make sense?